src/lib/routes/generate-answer-route.ts

```diff
@@ -70,18 +71,27 @@ export async function registerGenerateAnswerRoute(
 						temperature,
 						includeSummary: include_summary_in_response_generation,
 					});
-
-					const llm = new OpenAIClient(OPENAI_KEY);
 					let generatedAnswer: string = "";
-					const stream = await llm.requestResponseStream(
-						chatCompletionRequest,
-						(delta) => {
-							generatedAnswer += delta;
-						},
-					);
-
-					await reply.status(201).send(stream);
+					try {
+						const llm = new OpenAIClient(OPENAI_KEY);
+						const stream = await llm.requestResponseStream(
+							chatCompletionRequest,
+							(delta) => {
+								generatedAnswer += delta;
+							},
+						);
 
+						await reply.status(201).send(stream);
+					} catch (error: unknown) {
+						if (error instanceof Error) {
+							throw new OpenAIError("Failed to create completion", {
+								endpoint: "chat/completions",
+								status: 500,
+								statusText: error.message,
+							});
+						}
+						throw error;
+					}
```

Comment:
@ff6347  as just discussed in slack with @Jaszkowic , maybe we can write something (the error itself? a generic message like `openai api error`? idk) to the error column of user_requests, as otherwise we might have empty entries in the column `generated_answer` (as we throw an error here which will skip the supabase request below) and wonder what happened

Reply:
That would be good, yes.